go/worker/keymanager/worker.go

```diff
@@ -175,21 +175,18 @@ func (w *Worker) callLocal(ctx context.Context, data []byte) ([]byte, error) {
 }
 
 func (w *Worker) updateStatus(status *api.Status, startedEvent *host.StartedEvent) error {
-	var initOk bool
-	defer func() {
-		if !initOk {
-			// This is likely a new key manager that needs to replicate.
-			// Send a node registration anyway, so that other nodes know
-			// to update their access control.
-			w.roleProvider.SetAvailable(func(n *node.Node) error {
-				rt := n.AddOrUpdateRuntime(w.runtime.ID())
-				rt.Version = startedEvent.Version
-				rt.ExtraInfo = nil
-				rt.Capabilities.TEE = startedEvent.CapabilityTEE
-				return nil
-			})
-		}
-	}()
+	if !w.initialSyncDone {
+		// Send an initial node registration without any extra info, so that
```

Comment:
Can you explain a bit why this helps? So previously this would be called after init fails and now it is called immediately. But `SetAvailable` does not block until everyone has updated their access control lists so I'm not sure why calling it earlier would make a difference? If there is an observed difference it could be a timing quirk -- e.g., that this works because there is enough delay in trying to initialize the key manager, but it doesn't seem robust?

If this is actually problematic, wouldn't it be better that if `updateStatus` fails, but both "status" and "startedEvent" are available, the key manager worker loop would create a ticker that would retry key manager initialization? Of course you could have both, the change you implemented here (to release registration earlier) and also retry in case there are no policy updates, but initializing the key manager at a slightly later time would work due to propagation delay?

Otherwise could this cause the key manager to never initialize due to unfortunate timing and no new status updates?

Reply:
> But SetAvailable does not block until everyone has updated their access control lists so I'm not sure why calling it earlier would make a difference? If there is an observed difference it could be a timing quirk -- e.g., that this works because there is enough delay in trying to initialize the key manager, but it doesn't seem robust?

iirc the runtime does have some retry logic so this did always work, although I agree that in practice it could still fail.

> key manager worker loop would create a ticker that would retry key manager initialization

yeah i think this is a more proper solution, the keymanager should retry initialization in case `status`  and `startedEvent` are available and the initialization has not yet succeed.

> Of course you could have both

I think i will revert the current change just so I don't have to deal with fixing the initialization epochs in the e2e tets where mock epochtime is used

Reply:
done